### PR TITLE
Docs: Configure security hardening

### PR DESCRIPTION
**Pull request by KristianGrafana**
What is this feature?
This PR adds documentation for how to configure additional security hardening to the Grafana instance.

Why do we need this feature?
So Grafana admins know how to further secure the instance to protect against certain attacks (XSS and CSRF mainly). 

Who is this feature for?
Grafana admins and/or security engineers.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-security-hardening.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+# Set Content Security Policy template used when adding the Content-Security-Policy header to your requests.
+# $NONCE in the template includes a random nonce.
+# $ROOT_PATH is server.root_url without the protocol.
+content_security_policy_template = """script-src 'self' 'unsafe-eval' 'unsafe-inline' 'strict-dynamic' $NONCE;object-src 'none';font-src 'self';style-src 'self' 'unsafe-inline' blob:;img-src * data:;base-uri 'self';connect-src 'self' grafana.com ws://$ROOT_PATH wss://$ROOT_PATH;manifest-src 'self';media-src 'none';form-action 'self';"""
+```
+
+## Additional security hardening
+
+### Hide version
```
Comment (mgyongyosi) [inline docs/sources/setup-grafana/configure-security/configure-security-hardening.md:76]: I think the titles should be swapped: L76 with L88.

File: docs/sources/setup-grafana/configure-security/configure-security-hardening.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+> **Note:** Grafana must use HTTPS in order for the below configurations to work properly.
+
+### Secure cookie attribute
+
+Set the _cookie_secure_ option to `true`. This will enforce clients to set and send the cookie over a (valid HTTPS) secure connection to help mitigate against some MITM-attacks.
+
+Example:
+
+```toml
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-security-hardening.md:24]: which toml file?

File: docs/sources/setup-grafana/configure-security/configure-security-hardening.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+# set to true if you host Grafana behind HTTPS. default is false.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-security-hardening.md:25]:
```suggestion
# Set to true if you host Grafana behind HTTPS. The default value is false.
```

File: docs/sources/setup-grafana/configure-security/configure-security-hardening.md (hunk 2 above)
Comment (KristianGrafana) [reply docs/sources/setup-grafana/configure-security/configure-security-hardening.md:26]: I added a note in the beginning regarding which file to be modified.

File: docs/sources/setup-grafana/configure-security/configure-security-hardening.md (hunk 1 above)
Comment (KristianGrafana) [reply docs/sources/setup-grafana/configure-security/configure-security-hardening.md]: Thanks, good catch, fixed in https://github.com/grafana/grafana/pull/58738/commits/c401ffc47c9961af4291d05807759b6aef047786

File: docs/sources/setup-grafana/configure-security/configure-security-hardening.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  - /docs/grafana/latest/setup-grafana/configure-security/configure-security-hardening/
+description: Security hardening enables you to apply additional security which might stop certain vulnerabilities from being exploited by a malicious attacker. 
+title: Configure security hardening
+---
+
+# Configure  security hardening
+
+Security hardening enables you to apply additional security which might stop certain vulnerabilities from being exploited by a malicious attacker. 
+
+> **Note:** These settings will be found in the [grafana.ini configuration file]({{< relref "../configure-grafana/#configuration-file-location" >}}). After the file has been edited, the Grafana server needs to be restarted in order to apply the changes.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-security-hardening.md:12]:
```suggestion
> **Note:** These settings are available in the [grafana.ini configuration file]({{< relref "../configure-grafana/#configuration-file-location" >}}). To apply changes to the configuration file, restart the Grafana server.
```

**Pull request closed without merging**